### ISSUE TITLE
Move Eigen.hpp out of prim

### ISSUE THREAD

**Issue by SteveBronder** [stan-dev/math#3128]
## Description
Right now most files include `stan/math/prim/fun/Eigen.hpp` before everything else. 

```
#include <stan/math/prim/fun/Eigen.hpp>
#include <stan/math/rev/meta.hpp>
#include <stan/math/rev/core.hpp>
#include <stan/math/rev/fun/value_of.hpp>
```

But our rules for fwd and rev are the you should always include fwd/rev files before including anything from prim. So we should move Eigen.hpp up to the `math` folder so it is out of prim.

#### Current Version:
v4.9.0

**Comment by WardBrian** [comment]
I think this is a good idea, but we should probably leave `Eigen.hpp` in prim/fun, that includes whatever the new location is, because it _is_ used outside of just this repo:

https://github.com/search?q=org%3Astan-dev+stan%2Fmath%2Fprim%2Ffun%2FEigen.hpp+NOT+repo%3Astan-dev%2Fmath&type=code

**Comment by SteveBronder** [comment]
Yes that's reasonable. We can have the prim/fun file just include the other file
